packages/js/product-editor/src/products-app/product-edit/index.tsx

```diff
@@ -0,0 +1,160 @@
+/**
+ * External dependencies
+ */
+import { DataForm, isItemValid } from '@wordpress/dataviews';
+import type { Form } from '@wordpress/dataviews';
+import { createElement, useState, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import classNames from 'classnames';
+import {
+	// @ts-expect-error missing types.
+	__experimentalHeading as Heading,
+	// @ts-expect-error missing types.
+	__experimentalText as Text,
+	// @ts-expect-error missing types.
+	__experimentalHStack as HStack,
+	// @ts-expect-error missing types.
+	__experimentalVStack as VStack,
+	FlexItem,
+	Button,
+} from '@wordpress/components';
+// @ts-expect-error missing types.
+// eslint-disable-next-line @woocommerce/dependency-group
+import { privateApis as editorPrivateApis } from '@wordpress/editor';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+import { productFields } from '../product-list/fields';
+
+const { NavigableRegion } = unlock( editorPrivateApis );
+
+const form: Form = {
+	type: 'panel',
+	fields: [ 'name', 'status' ],
+};
+
+type ProductEditProps = {
+	subTitle?: string;
+	className?: string;
+	hideTitleFromUI?: boolean;
+	actions?: React.JSX.Element;
+	postType: string;
+	postId: string;
+};
+
+export default function ProductEdit( {
+	subTitle,
+	actions,
+	className,
+	hideTitleFromUI = true,
+	postType,
+	postId = '',
+}: ProductEditProps ) {
+	const classes = classNames( 'edit-product-page', className, {
+		'is-empty': ! postId,
+	} );
+	const ids = useMemo( () => postId.split( ',' ), [ postId ] );
+	const { initialEdits } = useSelect(
+		( select ) => {
+			return {
+				initialEdits:
+					ids.length === 1
+						? select( 'wc/admin/products' ).getProduct( ids[ 0 ] )
+						: null,
+			};
+		},
+		[ postType, ids ]
+	);
+	const [ edits, setEdits ] = useState( {} );
+	const itemWithEdits = useMemo( () => {
+		return {
+			...initialEdits,
+			...edits,
+		};
+	}, [ initialEdits, edits ] );
```

Comment:
This `edit` logic feels quite odd to me, but I copied it from the `posts` example in GB for now.
I did prefer to rely on the entity data store for this, the only caveat to that is how it will behave with multi-edit.